web/libs/editor/src/components/Comments/Comment/CommentForm.tsx

```diff
@@ -114,36 +127,69 @@ export const CommentForm: FC<CommentFormProps> = observer(({ commentStore, annot
-  const { text = "", regionRef } = currentComment || {};
+  const { text = "", regionRef, classifications } = currentComment || {};
   const { region, result } = regionRef || {};
   const linking = !!linkingComment && currentLinkingComment === linkingComment && globalLinking;
   const hasLinkState = linking || region;
+  const selections = taxonomyPathsToSelectedItems(classifications?.default?.values);
+  const classificationsItems = parseCommentClassificationConfig(commentStore.commentClassificationConfig);
+
+  const Buttons = () => (
+    <Elem name="buttons">
+      {!region && (
+        <Tooltip title="Link to..." mouseEnterDelay={TOOLTIP_DELAY}>
+          <Elem name="action" tag="button" mod={{ highlight: linking }} onClick={linkToHandler}>
+            <IconCommentLinkTo />
+          </Elem>
+        </Tooltip>
+      )}
+      <Elem name="action" tag="button" type="submit">
+        <IconSend />
+      </Elem>
+    </Elem>
+  );
 
   return (
     <Block ref={formRef} tag="form" name="comment-form-new" mod={{ inline, linked: !!region }} onSubmit={onSubmit}>
-      <TextArea
-        actionRef={actionRef}
-        name="comment"
-        placeholder="Add a comment"
-        value={text}
-        rows={ROWS}
-        maxRows={MAX_ROWS}
-        onInput={updateComment}
-        onSubmit={inline ? onSubmit : undefined}
-        onBlur={clearTooltipMessage}
-      />
-      <Elem tag="div" name="actions">
-        {!region && (
-          <Tooltip title="Link to..." mouseEnterDelay={TOOLTIP_DELAY}>
-            <Elem name="action" tag="button" mod={{ highlight: linking }} onClick={linkToHandler}>
-              <IconCommentLinkTo />
-            </Elem>
-          </Tooltip>
-        )}
-        <Elem name="action" tag="button" type="submit">
-          <IconSend />
-        </Elem>
+      <Elem name="text-row">
+        <TextArea
+          actionRef={actionRef}
+          name="comment"
+          placeholder="Add a comment"
+          value={text}
+          rows={ROWS}
+          maxRows={MAX_ROWS}
+          onInput={updateComment}
+          onSubmit={inline ? onSubmit : undefined}
+          onBlur={clearTooltipMessage}
+        />
+        {classificationsItems.length === 0 && <Buttons />}
       </Elem>
+      {classificationsItems.length > 0 && (
+        <Elem name="classifications-row">
+          <Elem name="category-selector">
+            <Taxonomy
+              selected={selections}
+              items={classificationsItems}
+              onChange={async (_, values) => {
+                const newClassifications =
+                  values.length > 0
+                    ? {
+                        default: {
+                          type: "taxonomy",
+                          values,
+                        },
+                      }
+                    : null;
+                updateCommentClassifications(newClassifications);
+              }}
+              options={{ pathSeparator: "/", showFullPath: true }}
```

Comment:
Given that this has inline definitions for both options and onChange, if the underlying component uses any memoization for performance on lengthy lists or deep tree rendering, this can cause a lot of unnecessary re-renders for sibling or parent change which would not warrant this component to need to update.

One way to avoid that is to pull the defined onChange as a useCallback in the component with `updateCommentClassifications` as its only dependency:

```
...
  const updateCommentClassifications = useCallback(
    (classifications: object | null) => {
      const currentComment = getCurrentComment();
      currentComment.setClassifications(classifications);
    },
    [commentStore, annotationStore],
  );

  const taxonomyOnChange = useCallback(async (_, values) => {
    const newClassifications =
      values.length > 0
        ? {
            default: {
              type: "taxonomy",
              values,
            },
          }
        : null;
    updateCommentClassifications(newClassifications);
  }, [updateCommentClassifications]);
```


Also lift the options `{ pathSeparator: "/", showFullPath: true }` to a module scope (outside the component) const definition:

```
...
const TOOLTIP_DELAY = 0.8;
const TAXONOMY_OPTIONS = { pathSeparator: "/", showFullPath: true };

export const CommentForm: FC<CommentFormProps> = observer(({ commentStore, annotationStore, inline = true }) => {
...
```

```suggestion
            <Taxonomy
              selected={selections}
              items={classificationsItems}
              onChange={taxonomyOnChange}
              options={TAXONOMY_OPTIONS}
```